substrate/frame/system/src/extensions/check_weight.rs

```diff
@@ -50,8 +50,18 @@ where
 	) -> Result<(), TransactionValidityError> {
 		let max = T::BlockWeights::get().get(info.class).max_extrinsic;
 		match max {
-			Some(max) if info.weight.any_gt(max) =>
-				Err(InvalidTransaction::ExhaustsResources.into()),
+			Some(max) if info.weight.any_gt(max) => {
+				log::debug!(
+					target: LOG_TARGET,
+					"Extrinsic weight ({}, {}) is greater than the max extrinsic weight ({}, {})",
```

Comment:
I think the `Display` impl for `Weight` is actually usable here.

Reply:
Nice!